coopr-rest-client/src/main/java/co/cask/coopr/client/ClusterClient.java

```
@@ -0,0 +1,135 @@
+/*
+ * Copyright © 2012-2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package co.cask.coopr.client;
+
+import co.cask.coopr.cluster.ClusterSummary;
+import co.cask.coopr.http.request.AddServicesRequest;
+import co.cask.coopr.http.request.ClusterCreateRequest;
+import co.cask.coopr.http.request.ClusterStatusResponse;
+import com.google.gson.JsonObject;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * The client API for manage clusters.
+ */
+public interface ClusterClient {
+
+  /**
+   * Provides a summary of details about all clusters visible to a user.
+   * If there are no visible clusters, returns empty list.
+   *
+   * @return list of {@link co.cask.coopr.cluster.ClusterSummary} objects
+   */
+  List<ClusterSummary> getClusters();
```

general comment about these calls is what happens under error conditions, for example if the API is not accessible?  Seems like we should throw an exception in that scenario, which would require a change to all these methods.